app/models/compute_resources/foreman/model/libvirt.rb

```diff
@@ -14,7 +14,35 @@ def self.available?
       Fog::Compute.providers.include?(:libvirt)
     end
 
-    # Some getters/setters for the attrs Hash
+    def self.firmware_types
+      {
+        "efi" => N_("EFI"),
+        "bios" => N_("BIOS"),
+      }.freeze
+    end
+
+    def self.os_loader_defaults
+      {
+        secure: "no",
```

Comment:
While testing secure boot with Fedora 39, I discovered that when using `shimx64.efi`, the secure boot is always enabled, ignoring the `loader secure=no`.

[From the docs:](https://libvirt.org/formatdomain.html)
> Attribute secure can be used to tell the hypervisor that the firmware is capable of Secure Boot feature. It cannot be used to enable or disable the feature itself in the firmware.

So now I'm not sure how to handle this in UI. Should we keep it as it is?

Reply:
If you use `shimx64.efi` as a bootloader and set the  `secure=no` in UI form, the VM will start with enabled secure boot.

It might be confusing for the users in the same way as it was for me.
I think the label should be `Secure Boot Capable` instead of `Secure Boot`.
Or add some help text explaining what is it exactly doing.

Reply:
As the docs says, Secure Boot is not enabled by just booting `shimx64.efi`. This is basically configured in the firmware.

IIRC most distributions always boot via `shimx64.efi`. If SecureBoot is not enabled in firmware, shim simply loads the bootloader (GRUB2) and that's it, no signature check at all.

According to https://libvirt.org/kbase/secureboot.html the interesting option is:
```
  <firmware>
    <feature enabled='yes' name='secure-boot'/>
    <feature enabled='yes' name='enrolled-keys'/>
  </firmware>
```

We would need to check what the actual fog-libvirt plugin (I assume) is doing when talking to libvirt API and to make sure, that corresponding (XML) options are provided when Secure Boot is enabled in the UI.

Reply:
Oh, so enabling secure_boot in UI should manage these firmware features ... I was thinking about them, where they are coming from; I thought they are set by libvirt itself, not by the fog. Hmm, will take a look at it.

Reply:
I had a quick look and I think here is a good point to start: https://github.com/fog/fog-libvirt/blob/21d1bf827c03ae79e1f5a10118c2b41fb80e0be7/lib/fog/libvirt/models/compute/server.rb#L271

Reply:
Looks like there was already some work done by you guys https://github.com/fog/fog-libvirt/pull/134 (at least adding UEFI firmware).

Reply:
yop, this is my follow up PR https://github.com/fog/fog-libvirt/pull/141

Reply:
So just for clarification (I didn't test by myself):

Is SB enabled or disabled depending in the firmware via `secure` variable in combination with https://github.com/fog/fog-libvirt/pull/134 or is it always enabled even if `secure=no`?

Reply:
It's always enabled. `secure=no` is just telling Libvirt "hey, this efi is capable of secure boot", nothing else.

Reply:
Ok, then libvirt's documentation is a bit confusing:

According to https://libvirt.org/formatdomain.html#bios-bootloader (as already stated in https://github.com/fog/fog-libvirt/issues/128) the following enables UEFI+SB:
```
<!-- QEMU with automatic UEFI firmware and secure boot -->
...
<os firmware='efi'>
  <type>hvm</type>
  <loader secure='yes'/>
  <boot dev='hd'/>
</os>
...
```
But it *doesn't* mean, that `<loader secure='no'/>` would lead to UEFI without SB, correct?

And honestly I also do not fully understand why libvirt should be aware of the SB state. At the end it depends on the user to provide fitting boot files for successful boot up when SB is enabled. Maybe for other software running on top like `virt-manager`. Anyway ...

We should then follow https://libvirt.org/kbase/secureboot.html and provide (as written above) to explicitly enable or disable SB depending on the user's decision in the UI (and provide corresponding `secure`  value in addition for whatever it is good for).

Reply:
> But it doesn't mean, that <loader secure='no'/> would lead to UEFI without SB, correct?

Exactly, that's why I started the discussion about the UI label. Secure Boot field in Foreman should be for enabling/disabling secure boot, and not only setting the `<loader secure='no'/>`.